### PR TITLE
feat: added continents function #984

### PR DESCRIPTION
**Pull request by charan-003**
- Added: A new continent() function to determine the continent for a given country.
- Fallback: Implemented a fallback to return "Unknown" for unmapped countries.

- Updates to Codebase:
- Modified location.cpp and location_data.h to include continent-related logic.

- Tests:
- Added tests in location_test.cpp under the new LocationContinentTest suite:
- shouldGenerateCorrectContinentForKnownCountry
- shouldReturnUnknownForUnmappedCountry
- shouldGenerateRandomContinent

### CODE REVIEW

File: tests/modules/location_test.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #include <string>
 #include <string_view>
 #include <vector>
+#include <iostream>
```
Comment (cieslarmichal) [inline tests/modules/location_test.cpp:6]: Remove

File: include/faker-cxx/location.h [hunk 2]
```diff
@@ -21,6 +21,23 @@ namespace faker::location
...
+ *
+ * @returns Continent name or "Unknown" if the country is not mapped.
+ *
+ * @code
+ * faker::location::continent("Poland") // "Europe"
+ * faker::location::continent() // Random continent
+ * @endcode
+ */
+FAKER_CXX_EXPORT std::string_view continent(std::string_view country = "");
+FAKER_CXX_EXPORT void validateCountryMapping();
```
Comment (cieslarmichal) [inline include/faker-cxx/location.h:37]: Remove, it should be only continent function in header file added

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Review by charan-003 — COMMENTED** [review]
removed unused methods

File: src/modules/location_data.h [hunk 3]
```diff
@@ -23,6 +23,268 @@ struct CountryAddressesInfo
     std::span<const std::string_view> states;
 };
 
+// adding continents data
+
+
```
Comment (cieslarmichal) [inline src/modules/location_data.h:28]: remove comment and empty lines

File: src/modules/location.cpp [hunk 4]
```diff
@@ -88,6 +90,34 @@ CountryAddressesInfo getAddresses(const Locale& locale)
...
+        //std::cout << "Found Continent: " << it->second << " for Country: " << selectedCountry << std::endl;
+        return it->second; // Return the corresponding continent
+    }
+    //std::cout << "Country Not Found: " << selectedCountry << ", Returning: Unknown" << std::endl;
+    return "Unknown"; // Fallback for unmapped countries
+}
+
+
+
+
```
Comment (cieslarmichal) [inline src/modules/location.cpp:119]: remove empty lines

File: src/modules/location.cpp [hunk 5]
```diff
@@ -88,6 +90,34 @@ CountryAddressesInfo getAddresses(const Locale& locale)
...
+    //std::cout << "Selected Country: " << selectedCountry << std::endl;
+
+    auto it = countryToContinent.find(selectedCountry);
+    if (it != countryToContinent.end())
+    {
+        //std::cout << "Found Continent: " << it->second << " for Country: " << selectedCountry << std::endl;
+        return it->second; // Return the corresponding continent
+    }
+    //std::cout << "Country Not Found: " << selectedCountry << ", Returning: Unknown" << std::endl;
```
Comment (cieslarmichal) [inline src/modules/location.cpp:113]: remove comments please

File: include/faker-cxx/location.h [hunk 6]
```diff
@@ -21,6 +21,20 @@ namespace faker::location
...
+/**
+ * @brief Returns the continent for a given country or a random continent if no country is specified.
+ *
+ * @param country The country name. Defaults to an empty string, which means a random country is selected.
+ *
+ * @returns Continent name or "Unknown" if the country is not mapped.
+ *
+ * @code
+ * faker::location::continent("Poland") // "Europe"
+ * faker::location::continent() // Random continent
```
Comment (cieslarmichal) [inline include/faker-cxx/location.h:33]: replace `Random continent` with actual random continent (any of them)

**Review by cieslarmichal — APPROVED** [review]

**Review by charan-003 — COMMENTED** [review]
removed comments and empty spaces, also added feature for random continent

File: include/faker-cxx/location.h [hunk 7]
```diff
@@ -21,6 +21,22 @@ namespace faker::location
...
+/**
+ * @brief Returns the continent for a given country or a random continent if no country is specified.
+ *
+ * @param country The country name. Defaults to an empty string, which means a random continent is selected.
+ *
+ * @returns Continent name or "Unknown" if the country is not mapped.
+ *
+ * @code
+ * faker::location::continent("Poland") // Europe
+ * faker::location::continent()
```
Comment (cieslarmichal) [inline include/faker-cxx/location.h:33]: Write some continent in comment like Africa

File: src/modules/location.cpp [hunk 8]
```diff
@@ -1,17 +1,21 @@
...
+#include "faker-cxx/types/locale.h"
 #include "common/algo_helper.h"
 #include "common/format_helper.h"
 #include "faker-cxx/helper.h"
 #include "faker-cxx/number.h"
 #include "faker-cxx/person.h"
-#include "faker-cxx/types/locale.h"
 #include "faker-cxx/types/precision.h"
 #include "location_data.h"
+#include <iostream>
```
Comment (cieslarmichal) [inline src/modules/location.cpp:17]: Remove please

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Review by charan-003 — COMMENTED** [review]
.

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**